RallyHereIntegration/Source/RallyHereIntegration/Private/RH_FriendSubsystem.cpp

```diff
@@ -34,28 +34,56 @@ void URH_FriendSubsystem::Initialize()
 	FriendsListPoller = FRH_PollControl::CreateAutoPoller();
 	BlockedListPoller = FRH_PollControl::CreateAutoPoller();
 
-	StartFriendsRefreshTimer();
-	StartBlockedRefreshTimer();
-
 	InitPropertiesWithDefaultValues();
 }
 
 void URH_FriendSubsystem::Deinitialize()
 {
 	UE_LOG(LogRallyHereIntegration, Verbose, TEXT("[%s]"), ANSI_TO_TCHAR(__FUNCTION__));
-	Super::Deinitialize();
 	InitPropertiesWithDefaultValues();
+
 	StopFriendsRefreshTimer();
 	StopBlockedRefreshTimer();
 
 	FriendsListPoller.Reset();
 	BlockedListPoller.Reset();
+
+	if (GetLocalPlayerSubsystem()->GetPlayerNotifications() != nullptr)
+	{
+		GetLocalPlayerSubsystem()->GetPlayerNotifications()->OnNotificationStreamedByAPI.FindOrAdd(TEXT("friends")).AddUObject(this, &URH_FriendSubsystem::HandleNotification);
+		GetLocalPlayerSubsystem()->GetPlayerNotifications()->OnNotificationStreamedByAPI.FindOrAdd(TEXT("blocked")).AddUObject(this, &URH_FriendSubsystem::HandleNotification);
+	}
+
+	Super::Deinitialize();
 }
 
-void URH_FriendSubsystem::OnUserChanged()
+void URH_FriendSubsystem::OnUserChanged(const FGuid& OldPlayerUuid, class URH_PlayerInfo* OldLocalPlayerInfo)
```

Comment:
Add notification binding, and also restart poll timers upon user change, to allow data to be flushed through quickly.